### PR TITLE
Improved Wording of HATEOAS Rule

### PR DESCRIPTION
**Pull request by fmueller**
I made headline consistent with 'MUST use rest maturity level 2' and I made wording a bit more clear after feedback from internal and external.

### CODE REVIEW

File: hyper-media/Hypermedia.md [hunk 1]
```diff
@@ -12,19 +12,24 @@ You can see this expressed by many rules throughout these guidelines, e.g.:
...
+But you should be aware of the shortcomings in our setup.
 
-Because we are following API First principles, HATEOAS brings nothing new to the table in terms of API self-descriptiveness. Furthermore, generic HATEOAS clients which crawl and use APIs on their own are only a theoretical concept so far.
-Our whole internal tooling around APIs like Twintip isn't adjusted for HATEOAS neither. For now, we have not
-seen a good reason to implement HATEOAS in an API.
+Because we are following API First principles, HATEOAS brings nothing new to the table in terms of API self-descriptiveness:
+a client finds necessary links in the API definition anyway. Furthermore, generic HATEOAS clients which crawl and
+use APIs on their own are only a theoretical concept so far. Our whole internal tooling around APIs like Twintip isn't
+adjusted for HATEOAS neither. For now, we have not seen a good reason to implement HATEOAS in an API as we are also mostly
+doing service-2-service communication in the backend. We still think that it could be useful in frontend and human-facing services.
```
Comment (ePaul) [inline hyper-media/Hypermedia.md]: Can we write this out as "service-to-service"? In an abbreviation like S2S I understand it, but here it sounds like is just to increase the geek factor.

Comment (fmueller) [inline hyper-media/Hypermedia.md]: Ok, already pushed the change for this.

**Comment by ePaul** [comment]
:+1:

**Comment by fmueller** [comment]
👍

File: hyper-media/Hypermedia.md [hunk 2]
```diff
@@ -14,13 +14,13 @@ Although this is not HATEOAS, it should not prevent you from designing proper li
...
-- Generic HATEOAS clients which need no prior knowledge about APIs and explore API capabilities based on hypermedia information provided, is a theoretical concept that we haven't seen working in practise and does not fit to our SOA set-up. Our whole internal tooling around APIs like Twintip isn't adjusted for HATEOAS neither.
+- We follow API First principle with APIs explicitly defined outside the code with standard specification language. HATEOAS does not really add value for SOA client engineers in terms of API self-descriptiveness: a client anyway finds necessary links and description how to use methods on endpoints depending on its state in the API definition.
+- Generic HATEOAS clients which need no prior knowledge about APIs and explore API capabilities based on hypermedia information provided, is a theoretical concept that we haven't seen working in practise and does not fit to our SOA set-up. Our whole internal tooling around APIs like OpenAPI isn't adjusted for HATEOAS neither.
```
Comment (ePaul) [inline hyper-media/Hypermedia.md]: OpenAPI isn't really an example for internal tooling. Maybe this:

> The OpenAPI description format (and tooling based on OpenAPI) doesn't provide great support for HATEOAS either.

Comment (tkrop) [inline hyper-media/Hypermedia.md]: great -> sufficient | good

Comment (tfrauenstein) [inline hyper-media/Hypermedia.md]: 👍

**Comment by fmueller** [comment]
@ePaul I've added your suggestion. Good one //cc @tfrauenstein @tkrop

**Comment by whiskeysierra** [comment]
:+1:

**Comment by tfrauenstein** [comment]
👍

**Comment by ePaul** [comment]
:+1:

**Pull request merged**